specification/snapshot/restore/SnapshotRestoreRequest.ts

```diff
@@ -83,7 +83,7 @@ export interface Request extends RequestBase {
      */
     wait_for_completion?: boolean
   }
-  body: {
+  body?: {
```

Comment:
YAML test that restores a snapshot without a body: https://github.com/elastic/elasticsearch/blob/a82b5cfb0ff7d22b95c56a8e7dcd7c5f3ba4ff24/modules/repository-gcs/src/yamlRestTest/resources/rest-api-spec/test/repository_gcs/20_repository.yml#L139-L144